### PR TITLE
Resolve conflicts, check integrity, return winners

### LINKED ISSUE #1: OrganismBoard should also keep track of the number of points scored
The `OrganismBoard` class should also be able to keep track of the number of points scored by each player so far. Ultimately we should be able to determine who won the game by just looking at an instance of this class.

### PR DESCRIPTION
**Pull request by ggsun**
This PR adds board methods that resolves conflicts between adjacent elements in the correct order, checks the integrity of each organism, and determines the winner(s) of the game. Fixes #1.

### CODE REVIEW

File: organism/organism.py [hunk 1]
```diff
@@ -190,13 +253,108 @@ def find_players(self):
...
+        if self.n_organisms_to_win > 0:
+            for player, player_organisms in organisms.items():
+                if len(player_organisms) >= self.n_organisms_to_win:
+                    winners.append(player)
+
+        # Find players with tokens equal to or more than the number to win
+        if self.n_tokens_to_win > 0:
+            for player, n_tokens in self.tokens.items():
+                if n_tokens >= self.n_tokens_to_win:
+                    winners.append(player)
```
Comment (prismofeverything) [inline organism/organism.py]: This assumes that one player could win on tokens while another player wins on organisms, in the same turn. We could do that? I can imagine it happening though it seems like a corner case. 

The bigger condition is that a player loses if they suicide, even if the other player dies, which we should probably encode somehow.

File: organism/organism.py [hunk 2]
```diff
@@ -190,13 +253,108 @@ def find_players(self):
...
+                    if adjacent_state['element'] is not None:
+                        adjacent_player = adjacent_state['element']['player']
+                        adjacent_element_type = adjacent_state['element']['type']
+
+                        if adjacent_player != player and adjacent_element_type == element_heterarchy[element_type]:
+                            conflicts.append((adjacent_space, space))
+                            self.tokens[player] += 1
+
+        # Topologically sort the list of conflicts
+        sorted_conflicts = topological_sort(conflicts)
```
Comment (prismofeverything) [inline organism/organism.py]: Excellent use of topological sort.

File: organism/organism.py [hunk 3]
```diff
@@ -190,13 +253,108 @@ def find_players(self):
...
+                if set(element_types) != full_set:
+                    for space in organism:
+                        self._remove_element(space)
+
+                    n_removed_elements = len(organism)
+
+                    # Add tokens for all other players
+                    for p in self.find_players():
+                        if p != player:
+                            self.tokens[p] += n_removed_elements
```
Comment (prismofeverything) [inline organism/organism.py]: So this rewards tokens to all players besides the one whose incomplete organism was dissolved, which I believe we want to award them to the player whose turn it currently is (or was). This is fine for two player games but we probably want to fix it at some point. 

If it is the current player's organism that dissolves this turn we can award tokens this way.

**Comment by prismofeverything** [comment]
Does this approach extend to three-player games? Or beyond? I almost feel like once it is trained we should be able to use it in a three more more player game, but there are probably additional considerations I'm not thinking of.

**Comment by ggsun** [comment]
Hey @prismofeverything, thanks for the input! I wasn't thinking too much about the three-player mode when I implemented these, and there are a lot more gray areas to the rules than I imagined including some that you mentioned here. I think a prerequisite here would be having the Board hold data on the current player and the remaining organisms that the current player can "move".

File: organism/organism.py (hunk 1 above)
Comment (ggsun) [reply organism/organism.py]: I like the suicide = loss rule. I'm guessing you're not a big fan of tied games?

Comment (ggsun) [reply organism/organism.py]: I just realized you have a detailed win condition written down in the manual, my bad :/ I'll try to implement what you have written down there.

**Pull request merged**